src/common/meta/src/node_manager.rs

```diff
@@ -16,13 +16,27 @@ use std::sync::Arc;
 
 use api::region::RegionResponse;
 use api::v1::flow::{FlowRequest, FlowResponse};
-use api::v1::region::{InsertRequests, QueryRequest, RegionRequest};
+use api::v1::region::{InsertRequests, RegionRequest, RegionRequestHeader};
 pub use common_base::AffectedRows;
 use common_recordbatch::SendableRecordBatchStream;
+use datafusion_expr::LogicalPlan;
+use store_api::storage::RegionId;
 
 use crate::error::Result;
 use crate::peer::Peer;
 
+/// The query request to be handled by the RegionServer(Datanode).
+pub struct QueryRequest {
+    /// The header of this request. Often to store some context of the query. None means all to defaults.
+    pub header: Option<RegionRequestHeader>,
+
+    /// The id of the region to be queried.
+    pub region_id: RegionId,
+
+    /// The form of the query: a logical plan.
+    pub plan: LogicalPlan,
+}
```

Comment:
Putting the `QueryRequest` in the `common-meta` is weird. What about moving it to `common-query`?

Reply:
Can't do it because `RegionId` is in `store-api`, and `store-api` depends on `common-query`.

Reply:
Do we need to keep `region_id` in the `QueryRequest`? We can pass `region_id` to `handle_read()` like `handle_request()`.
https://github.com/GreptimeTeam/greptimedb/blob/a6a702de4e51e91d4066d5a8e916b2e99116316a/src/datanode/src/region_server.rs#L117-L121

Reply:
I think it's better to lessen the args of a function. The `QueryRequest` is directly used in `trait Datanode` defined in the same file, if `QueryRequest` is to be not placed in common-meta crate, so is that trait. I think we can do it in the future, to better organize the codes.

Reply:
moved to `common-query`